airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivity.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import java.time.Instant;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@ActivityInterface
+public interface AutoDisableConnectionActivity {
+
+  @Data
+  @NoArgsConstructor
+  @AllArgsConstructor
+  class AutoDisableConnectionActivityInput {
+
+    private UUID connectionId;
+
+    private Instant currTimestamp;
+
+  }
+
+  /**
+   * Disable a connection if no successful sync jobs in the last MAX_FAILURE_JOBS_IN_A_ROW job
+   * attempts or the last MAX_DAYS_OF_STRAIGHT_FAILURE days (minimum 1 job attempt): disable
+   * connection to prevent wasting resources
+   */
+  @ActivityMethod
+  void autoDisableFailingConnection(AutoDisableConnectionActivityInput input);
```

Comment:
Nit: Sorry I forget it before but it could be nice to return an object containing a boolean that will say if it got disabled or not. The main benefit will be to allow to see that directly in the UI.

Reply:
done